W-20179825 Refactor agent network documentation (#549)
mulesoft/docs-code-builder: modules/ROOT/pages/_partials/af-shared.adoc

// tag::agent-support[]
Agent assets must be A2A-compliant.
// end::agent-support[]
//

//
// tag::get-started[]

> Creating agent networks is multi-step process. Review these prerequisites and task overviews to ensure that you can create agent networks efficiently.

Reviewer: ```suggestion
Creating agent networks is a multi-step process. Review these prerequisites and task overviews to make sure you can create agent networks efficiently.
```

[[cli-requirements]]
=== CLI Requirements (Optional)

If you're using the Anypoint CLI to create agent network projects, ensure you meet these requirements.

* Node 20 (v20.19.4) or later. See http://Nodejs.org[Nodejs.org].
* Java 17 or later. Set the `JAVA_HOME` environment variable.
* xref:af-build-agent-networks-in-a-ci-cd-environment.adoc[Anypoint CLI Fabric Plugin]

[[overview]]
== Task Overview

[[step-1-setup]]
=== Step 1: Set Up Your Agent Network

. Use Anypoint Runtime Manager to set up a private space in CloudHub 2.0 or deployment target in Runtime Fabric.
+
See <<setup-space>>.
. Set up Ingress and egress Flex Gateways for the private space or deployment target.

  * If using Anypoint Code Builder, see <<setup-gateways>>.
  * If using the Anypoint CLI, see the xref:af-build-agent-networks-in-a-ci-cd-environment.adoc#agent-network-setup-gateways[setup gateways command reference].

[[step-2-create]]
=== Step 2: Create a Project

Now you're ready to create an agent network project. Choose one of the following methods:

> * xref:af-create-agent-network.adoc#create-dev-agent[Create a Network Using the MuleSoft Dev Agent]

Reviewer: I'm going to leave this one up to you because it's nitpicky but I just realized all these links are complete sentences, so periods at the end?

> * xref:af-create-agent-network.adoc#create-dev-agent[Create a Network Using the MuleSoft Dev Agent]

Reviewer: ```suggestion
* xref:af-create-agent-network.adoc#create-dev-agent[Create a Network Using MuleSoft Dev Agent]
```

* xref:af-create-agent-network.adoc#create-acb[Create a Network Using Anypoint Code Builder]
* xref:af-create-agent-network.adoc#create-cli[Create a Network Using the Anypoint CLI]

[[step-3-define]]
=== Step 3: Define Your Agent Network Specification

After you create your agent network project, configure `agent-network.yaml` and `exchange.json` to reflect the structure of your network.

> * xref:af-define-your-agent-network-specification.adoc#define-dev-agent[Define a Network Using the MuleSoft Dev Agent]

Reviewer: ```suggestion
* xref:af-define-your-agent-network-specification.adoc#define-dev-agent[Define a Network Using MuleSoft Dev Agent]
```

* xref:af-define-your-agent-network-specification.adoc#define-acb-ide[Define a Network Using Anypoint Code Builder or IDE]

[[step-4-publish]]
=== Step 4: Publish Your Agent Network Assets

Build and publish your agent network project as Anypoint Exchange assets. When you publish the agent network, an asset is created in Exchange for each broker, agent, and MCP server that's in your agent network.

> * xref:af-publish-agent-network-assets.adoc#publish-dev-agent[Publish Your Network Using the MuleSoft Dev Agent]

Reviewer: ```suggestion
* xref:af-publish-agent-network-assets.adoc#publish-dev-agent[Publish Your Network Using MuleSoft Dev Agent]
```

Reviewer: And then throughout, just get rid of "the" in front of MuleSoft Dev Agent.
